### PR TITLE
[MINOR][CONNECT][DOC] Add information on how to regenerate proto for python client

### PR DESCRIPTION
**Pull request by juliuszsompolski**
### What changes were proposed in this pull request?

Figuring out how to generate connect grpc proto in python was surprisingly hard to figure out for me (not knowing much about python development though), so adding it to the README.

### Why are the changes needed?

Improving internal documentation.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Not applicable.

### CODE REVIEW

File: connector/connect/README.md [hunk 1]
```diff
@@ -21,6 +21,17 @@ user experience across all languages. Please follow the below guidelines:
...
+It depends on
+```
+brew install bufbuild/buf/buf
+```
+and python dependencies from
+```
+pip install -r dev/requirements.txt
+```
+(specifically, install `black` and dependencies from "Spark Connect python proto generation plugin (optional)" section)
+
```
Comment (nija-at) [inline connector/connect/README.md:34]: I would've written it differently.

````suggestion
To generate the Python client code from the proto files,

First, make sure to have a Python environment with the installed dependencies:

```
pip install -r dev/requirements.txt
```

Install [buf](https://github.com/bufbuild/buf)

```
brew install bufbuild/buf/buf
```

Generate the Python files by running:

```
dev/connect-gen-protos.sh
```

````

File: connector/connect/README.md [hunk 2]
```diff
@@ -21,6 +21,17 @@ user experience across all languages. Please follow the below guidelines:
 
 Python-specific development guidelines are located in [python/docs/source/development/testing.rst](https://github.com/apache/spark/blob/master/python/docs/source/development/testing.rst) that is published at [Development tab](https://spark.apache.org/docs/latest/api/python/development/index.html) in PySpark documentation.
```
Comment (nija-at) [inline connector/connect/README.md:23]: Would it be useful to add a README in this folder and link to this README?

**Review by nija-at — APPROVED** [review]

**Review by grundprinzip — APPROVED** [review]

File: connector/connect/README.md (hunk 2 above)
Comment (juliuszsompolski) [reply connector/connect/README.md:23]: I think that folder is `rst` based and published to the web documentation and not markdown based, so I think README.md doesn't fit there.
But I was surprised that the documentation there doesn't mention getting dependencies from `dev/requirements.txt`. But I'm not really familiar with python dev and these docs, maybe someone more familiar should have a better idea what would be the best place for it? @HyukjinKwon ?

File: connector/connect/README.md [hunk 3]
```diff
@@ -21,6 +21,28 @@ user experience across all languages. Please follow the below guidelines:
...
+Install [buf](https://github.com/bufbuild/buf)
+
+```
+brew install bufbuild/buf/buf
+```
+
+Generate the Python files by running:
+
+```
+dev/connect-gen-protos.sh
```
Comment (MaxGekk) [inline connector/connect/README.md:43]: Don't know is it specific to me but this command fails on my site with error:
```
+ buf generate --debug -vvv
...
Failure: 403 Forbidden
```

A workaround is to fork https://github.com/HyukjinKwon/my-github-actions and run the workflow:
Screen Shot 2023-03-24 at 8 38 18 PM
and download the results:
Screen Shot 2023-03-24 at 8 38 45 PM

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @juliuszsompolski and @grundprinzip @nija-at for review.

File: connector/connect/README.md [hunk 4]
```diff
@@ -21,6 +21,28 @@ user experience across all languages. Please follow the below guidelines:
...
+
+
+```
+pip install -r dev/requirements.txt
+```
+
+Install [buf](https://github.com/bufbuild/buf)
+
+```
+brew install bufbuild/buf/buf
```
Comment (HyukjinKwon) [inline connector/connect/README.md:37]: and at https://spark.apache.org/docs/latest/api/python/development/contributing.html#prerequisite

File: connector/connect/README.md (hunk 3 above)
Comment (HyukjinKwon) [inline connector/connect/README.md:43]: This is actually documented at https://spark.apache.org/docs/latest/api/python/development/testing.html#running-tests-for-python-client

File: connector/connect/README.md [hunk 5]
```diff
@@ -21,6 +21,28 @@ user experience across all languages. Please follow the below guidelines:
 
 Python-specific development guidelines are located in [python/docs/source/development/testing.rst](https://github.com/apache/spark/blob/master/python/docs/source/development/testing.rst) that is published at [Development tab](https://spark.apache.org/docs/latest/api/python/development/index.html) in PySpark documentation.
 
+To generate the Python client code from the proto files:
+
+First, make sure to have a Python environment with the installed dependencies.
+Specifically, install `black` and dependencies from the "Spark Connect python proto generation plugin (optional)" section.
```
Comment (HyukjinKwon) [inline connector/connect/README.md:27]: no biggie but I suspect we won't necessarily need to mention this

**Review by HyukjinKwon — COMMENTED** [review]
LGTM. Let's clean up and dedup a little bit next time when we touch here.

File: connector/connect/README.md (hunk 3 above)
Comment (nija-at) [reply connector/connect/README.md:43]: @juliuszsompolski was not able to find them. So I suspect they are not in a very discoverable spot.

Dedupe in a follow up sounds good. We can also change these to be links to the correct place.

**Pull request closed without merging**